k-distribution/pl-tutorial/1_k/4_imp++/lesson_6/imp.k

```diff
@@ -6,14 +6,14 @@ module IMP-SYNTAX
                  | "++" Id
                  | "read" "(" ")"
                  | "-" Int
+                 | "(" AExp ")"               [bracket]
                  > AExp "/" AExp              [left, strict, division]
                  > AExp "+" AExp              [left, strict]
-                 | "(" AExp ")"               [bracket]
```

Comment:
Think about it. The previous order was saying that `()` couldn't appear under a `/`, which we obviously want to allow. Once I fixed the bug that was preventing the priority relation the definition had expressed from being applied, suddenly you had parsing errors because the grammar had been incorrectly specified. So I fixed it.